noir-projects/noir-protocol-circuits/crates/reset-kernel-lib/src/reset/key_validation_hint.nr

```diff
@@ -77,13 +77,13 @@ pub fn verify_reset_key_validation_requests<let REQUEST_LEN: u32, let NUM_HINTS:
     let mut num_propagated = 0;
     for i in 0..REQUEST_LEN {
         let hint_index = hint_indexes[i];
-        if hint_index == NUM_HINTS {
+        if (hint_index != NUM_HINTS) & (NUM_HINTS != 0) {
```

Comment:
This changes allows `NUM_HINTS` to be 0.

However, the following will still trigger error when generating vk:

```rust
if (hint_index == NUM_HINTS) | (NUM_HINTS == 0) {
  // propagate
} else {
  assert_eq((hints[hint_index].request_index, i);
}
```